pkg/controller/awsfederatedaccountaccess/awsfederatedaccountaccess_controller_test.go

```diff
@@ -0,0 +1,48 @@
+package awsfederatedaccountaccess
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+
+	"github.com/golang/mock/gomock"
+
+	"github.com/openshift/aws-account-operator/pkg/awsclient/mock"
+	"github.com/openshift/aws-account-operator/pkg/controller/testutils"
+
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type mocks struct {
+	mockCtrl *gomock.Controller
+}
+
+func setupDefaultMocks(t *testing.T, localObjects []runtime.Object) *mocks {
+	mocks := &mocks{
+		mockCtrl: gomock.NewController(t),
+	}
+
+	return mocks
+}
+
+func TestCheckAndDeletePolicy(t *testing.T) {
```

Comment:
We're currently only testing the happy path here, would be good to test a failed DeletePolicy call

Reply:
Still relevant